### PR TITLE
Ensure MaxConcurrentRequests is a positive number

### PR DESCRIPTION
**Pull request by Kahbazi**
@DylanDmitri

### CODE REVIEW

File: src/Middleware/RequestThrottling/src/RequestThrottlingMiddleware.cs [hunk 1]
```diff
@@ -35,6 +35,10 @@ public RequestThrottlingMiddleware(RequestDelegate next, ILoggerFactory loggerFa
             {
                 throw new ArgumentException("The value of 'options.MaxConcurrentRequests' must be specified.", nameof(options));
             }
+            if (_requestThrottlingOptions.MaxConcurrentRequests < 0)
+            {
+                throw new ArgumentException("The value of 'options.MaxConcurrentRequests' must be a positive integer.", nameof(options));
+            }
```
Comment (Eilon) [inline src/Middleware/RequestThrottling/src/RequestThrottlingMiddleware.cs]: I feel this is the wrong place to do validation. The setter on the property should throw for invalid values:

https://github.com/aspnet/AspNetCore/blob/master/src/Middleware/RequestThrottling/src/RequestThrottlingOptions.cs#L19

**Review by Eilon — DISMISSED** [review]

**Comment by DylanDmitri** [comment]
If `MaxConcurrentRequests` is negative (or zero), there's currently no feedback and the server never responds to any requests. This is BAD behavior and we want to prevent this. 

However, some tests rely on using 0 for `MaxConcurrentRequests` to make sure that nothing sneaks out of the queue. Adding the validation check breaks those tests, so it's all trickier than expected.

**Comment by Kahbazi** [comment]
The check allows `MaxConcurrentRequests` to be 0 so the tests will pass. Also if `MaxConcurrentRequests` is negative we have a exception anyway https://github.com/aspnet/AspNetCore/blob/master/src/Middleware/RequestThrottling/src/Internal/RequestQueue.cs#L23

File: src/Middleware/RequestThrottling/src/RequestThrottlingMiddleware.cs (hunk 1 above)
Comment (Tratcher) [reply src/Middleware/RequestThrottling/src/RequestThrottlingMiddleware.cs]: Note we don't currently use that pattern in this middleware (or any other middleware).

Recommendation: take this change as is and then follow up with a change to move this all to Options Validation.
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-2.2#options-validation

**Comment by DylanDmitri** [comment]
@Kahbazi good catch on the issue!

**Pull request merged**